### PR TITLE
SDI-505 Handle activities.activity.created event

### PR DESCRIPTION
**Pull request by steverendell**
correct event name, remove generic queue, use topic

### CODE REVIEW

File: src/main/resources/application-localstack.yml [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
   queues:
-    prisoner:
-      queueName: hmpps_prisoner_to_nomis_queue
-      dlqName: hmpps_prisoner_to_nomis_dlq
-      subscribeTopicId: domainevents
     visit:
       queueName: hmpps_prisoner_to_nomis_visit_queue
       dlqName: hmpps_prisoner_to_nomis_visit_dlq
+      subscribeTopicId: domainevents
+      subscribeFilter: "{\"eventType\":[\"prison-visit.booked\", \"prison-visit.changed\", \"prison-visit.cancelled\"]}"
```
Comment (steverendell) [inline src/main/resources/application-localstack.yml:8]: I changed the integration tests to send messages to the topic

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesApiServiceTest.kt [hunk 2]
```diff
@@ -114,7 +114,8 @@ internal class ActivitiesApiServiceTest {
             "endTime": "11:30",
             "daysOfWeek": ["Mon","Tue","Wed"]
           }
-        ]
+        ],
+        "startDate" : "2023-01-20"
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesApiServiceTest.kt:118]: now mandatory

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/integration/ActivityToNomisTest.kt [hunk 3]
```diff
@@ -22,10 +23,15 @@ class ActivityToNomisTest : SqsIntegrationTestBase() {
...
-      SendMessageRequest.builder().queueUrl(activityQueueUrl).messageBody(message).build()
+    awsSnsClient.publish(
+      PublishRequest.builder().topicArn(topicArn)
+        .message(activityMessagePayload("activities.activity-schedule.created", 12))
+        .messageAttributes(
+          mapOf(
+            "eventType" to MessageAttributeValue.builder().dataType("String")
+              .stringValue("activities.activity-schedule.created").build(),
+          )
+        ).build()
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/integration/ActivityToNomisTest.kt:34]: the message published to the topic is just the payload, so simpler than the boilerplate wrapper we sent to the queue

File: src/test/resources/application-test.yml [hunk 4]
```diff
@@ -17,31 +17,32 @@ api.base.url:
...
       dlqName: ${random.uuid}
+      subscribeTopicId: domainevents
+      subscribeFilter: "{\"eventType\":[\"activities.activity-schedule.created\"]}"
       dlqMaxReceiveCount: 1
       visibilityTimeout: 1
     sentencing:
       queueName: ${random.uuid}
       dlqName: ${random.uuid}
+      subscribeTopicId: domainevents
+      subscribeFilter: "{\"eventType\":[\"sentencing.sentence.adjustment.created\", \"sentencing.sentence.adjustment.updated\", \"sentencing.sentence.adjustment.deleted\"]}"
```
Comment (steverendell) [inline src/test/resources/application-test.yml:45]: would be nice to define these filters in one place 🤷

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/helpers/Messages.kt [hunk 5]
```diff
@@ -68,6 +40,9 @@ fun retryMessage() = """
       }
 """.trimIndent()
 
+fun incentiveMessagePayload(incentiveId: Long) =
+  """{"eventType":"incentives.iep-review.inserted", "additionalInformation": {"id":"$incentiveId"}}"""
+
 fun incentiveCreatedMessage(incentiveId: Long) = """
```
Comment (lbennett-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/helpers/Messages.kt:46]: Does this one go also - or is it still being used?

**Review by lbennett-moj — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/helpers/Messages.kt (hunk 5 above)
Comment (steverendell) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/helpers/Messages.kt:46]: still used by unit test

**Pull request merged**